.gitignore

```diff
@@ -1,5 +1,9 @@
+*build*/
 .*.swp
+*.bak
+*.diff
+*.json
+*.patch
```

Comment:
`*.diff` and `*.batch` are better added to your own local config (in `$HOME/.config/git/ignore`). It's a personal preference, not specific to this repository (I should probably remove `.*.swp` for the same reason).

For `*.json`, I would prefer being explicit and exclude `compile_commands.json`. If this project ever needs to ship a JSON file, we should be able to do it.

Similarly, I would prefer `build*/` to `*build*/` (but +1 for adding the trailing `/`)

Reply:
being at it I could remove the swp extension here? For the rest I'm going to do as you suggested (I also removed the bak one, and thank you for the global ignore suggestion, I didn't know! :P )

Reply:
Sorry for the late reply, was away. Yes, feel free to remove the swp extension.